### PR TITLE
feat(core)!: Introduce two new CRDs and update existing ones

### LINKED ISSUE #34: Cleanup code: Implement new CRD for Intent with ID, params

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes #34 

**Does this PR introduce a breaking change?**
Yes.

## Checklist

- [x] PR title follows the `<type>: <description>` convention
- [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
- [x] I have updated the [documentation](../docs) accordingly
- [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
- [ ] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Additional information for reviewer
**Hold off on merging this PR for now. The CRDs' `status` field is still under development and need to be finalized before this PR can be integrated.**

### CODE REVIEW

**Review by seungsoo-lee — APPROVED** [review]

File: api/v1/nimbuspolicy_types.go [hunk 1]
```diff
@@ -30,107 +24,21 @@ type NimbusSelector struct {
...
-type MatchCapability struct {
-	Capability string             `json:"capability,omitempty"`
-	FromSource []NimbusFromSource `json:"fromSource,omitempty"`
-}
-
-// FromSource defines a source path for directory-based policies
-type NimbusFromSource struct {
-	Path string `json:"path,omitempty"`
+	RuleAction string              `json:"action"`
+	Mode       string              `json:"mode,omitempty"`
```
Comment (shivaccuknox) [inline api/v1/nimbuspolicy_types.go:35]: Please add some description for each field

File: api/v1/securityintent_types.go [hunk 2]
```diff
@@ -7,119 +7,50 @@ import (
...
+	// Action defines how the security policy will be enforced.
+	Action string `json:"action"`
 
-// MatchSyscall defines a syscall for syscall policies
-type SecurityIntentMatchSyscall struct {
-	Syscalls   []string            `json:"syscalls,omitempty"`
-	FromSource []SyscallFromSource `json:"fromSource,omitempty"`
-}
+	// Mode defines the enforcement behavior of the intent.
+	// Defaults to best-effort.
```
Comment (shivaccuknox) [inline api/v1/securityintent_types.go:29]: Please mention "strict" here, and also what it means

Comment (anurag-rajawat) [reply api/v1/securityintent_types.go:29]: > Please mention "strict" here

We'll be updating our documentation to list all possible field values, so I don't think there's a need to mention `strict` here.

> what it means

 Mode defines the enforcement behavior of the intent.

**Comment by anurag-rajawat** [comment]
`ClusterNimbusPolicies` are temporarily not supported by the `nimbus-kubearmor` adapter.

**Comment by anurag-rajawat** [comment]
> ClusterNimbusPolicies are temporarily not supported by the nimbus-kubearmor adapter.

This issue has been fixed.

**Pull request merged**